specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/expressRouteCrossConnection.json

```diff
@@ -47,6 +47,13 @@
           },
           {
             "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "name",
+            "in": "query",
```

Comment:
pl use OData filters rather than individual query params

Reply:
@rkmanda Thanks for commenting. 
About using OData filter, should I add the following to the file? Should I make any backend changes or manifest changes for it to work as expected? It doesn't just work out of the box, right?

```
{
            "name": "$filter",
            "in": "query",
            "description": "An OData filter expression that filters elements in the collection.",
            "required": false,
            "type": "string"
}
```

We are in a difficult situation that our RDFE API is deprecating in 2 months, and this filtering stuff has to work. My "name" query parameter has special NRP handling already in place and it's working well. And it's already communicated to the customer. If the OData filter needs more NRP changes, we will definitely have no time for that. I'll add you to the escalation email thread for this. 

If the OData filter is a must-have, can I add the above code along with my current change, so I can at least have a safety guarantee?

Reply:
@jianzhao1992 can you also add me to the email thread?

Reply:
Hi @razvanbadea-msft I have added you to the email thread.

Reply:
What's the outcome/decision on this thread?

Reply:
@ms-bogdan-rotaru The theme of the email thread is to push for a quick resolution of this PR, nothing technical.

Reply:
Use of OData filters is the right thing to do and yes you should add the $filter query param and also have your service runtime implement the contract. https://learn.microsoft.com/en-us/odata/concepts/queryoptions-overview